pkg/scaling/resolver/scale_resolvers_test.go

```diff
@@ -592,3 +594,66 @@ func TestEnvWithRestrictSecretAccess(t *testing.T) {
 		})
 	}
 }
+
+func TestEnvWitWhitelistedNamespace(t *testing.T) {
+	tests := []struct {
+		name      string
+		expected  map[string]string
+		container *corev1.Container
+	}{
+		{
+			name: "env reference secret key",
+			container: &corev1.Container{
+				Env: []corev1.EnvVar{{
+					Name: envKey,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: secretKey,
+						},
+					},
+				}},
+			},
+			expected: map[string]string{envKey: secretData},
+		},
+		{
+			name: "env reference secret name",
+			container: &corev1.Container{
+				EnvFrom: []corev1.EnvFromSource{{
+					SecretRef: &corev1.SecretEnvSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secretName,
+						},
+					},
+				}},
+			},
+			expected: map[string]string{secretKey: secretData},
+		},
+	}
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: clusterNamespace,
+			Name:      secretName,
+		},
+		Data: map[string][]byte{secretKey: []byte(secretData)},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mock_secretNamespaceLister := mock_v1.NewMockSecretNamespaceLister(ctrl)
+			mock_secretNamespaceLister.EXPECT().Get(secretName).Return(secret, nil).AnyTimes()
+			mock_secretLister := mock_v1.NewMockSecretLister(ctrl)
+			mock_secretLister.EXPECT().Secrets(clusterNamespace).Return(mock_secretNamespaceLister).AnyTimes()
+			os.Setenv("KEDA_CLUSTER_OBJECT_NAMESPACE", "keda")
+			os.Setenv("KEDA_RESTRICT_SECRET_ACCESS", "true")
+			ctx := context.Background()
+			envMap, _ := resolveEnv(ctx, fake.NewClientBuilder().Build(), logf.Log.WithName("test"), test.container, clusterNamespace, mock_secretLister)
+			if diff := cmp.Diff(envMap, test.expected); diff != "" {
+				t.Errorf("Returned env map is different: %s", diff)
+			}
+		})
+	}
```

Comment:
Since the mock has no constraint or specific return value, what about having it out of the loop?

```suggestion
        ctrl := gomock.NewController(t)
	mock_secretNamespaceLister := mock_v1.NewMockSecretNamespaceLister(ctrl)
	mock_secretNamespaceLister.EXPECT().Get(secretName).Return(secret, nil).AnyTimes()
	mock_secretLister := mock_v1.NewMockSecretLister(ctrl)
	mock_secretLister.EXPECT().Secrets(clusterNamespace).Return(mock_secretNamespaceLister).AnyTimes()
	for _, test := range tests {
		test := test
		t.Run(test.name, func(t *testing.T) {
			os.Setenv("KEDA_CLUSTER_OBJECT_NAMESPACE", "keda")
			os.Setenv("KEDA_RESTRICT_SECRET_ACCESS", "true")
			ctx := context.Background()
			envMap, _ := resolveEnv(ctx, fake.NewClientBuilder().Build(), logf.Log.WithName("test"), test.container, clusterNamespace, mock_secretLister)
			if diff := cmp.Diff(envMap, test.expected); diff != "" {
				t.Errorf("Returned env map is different: %s", diff)
			}
		})
	}
```

Reply:
Thanks @tobotg for suggestion. Have addressed the same 😃

Reply:
@tobotg thanks for your suggestion. Have addressed the same 😃

Reply:
Thanks @Indresh2410 👍